source/ncengine/graphics2/diligent/resource/ShaderBindings.h

```diff
@@ -13,10 +14,20 @@ struct FrontendRenderState;
 class ShaderBindings
 {
     public:
+        /** @todo: 782 Fully parameterize names, indices, and size hints */
         explicit ShaderBindings(Diligent::IRenderDevice& device,
                                 Diligent::IDeviceContext& context,
-                                uint32_t maxTextures)
-            : m_globalSignature{device, context, maxTextures}
+                                uint32_t maxTextures,
+                                uint32_t initialMaterialSizeHint)
+            : m_globalSignature{device, context, maxTextures},
+              m_materialSignature{
+                context,
+                device,
+                "MaterialInstanceSignature",
+                "MaterialDataBuffer",
+                1,
```

Comment:
Let's move this binding index to 2. It makes sense to me to put ComponentResourceSignature at 1, because it's the second most general grouping.
Also, let's extract these to a constant and put them in `ShaderBindings`

Reply:
Updated to 2, but see the comment above about #782. I want to parameterize `ShaderBindings` with a struct to hold names/indices (or it could just be a bunch of constants). Those also feel like part of the interface and shouldn't be buried in diligent/backend, but probably in `ShaderTypes`. (??) Anyway, there is an issue to collect all of those variables together once this and your component resource are in.